test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextNoTrackingStreamsTests.cs

```diff
@@ -0,0 +1,450 @@
+            Assert.Equal(3, DefaultTrackingContext.Entities.ToList().Count);
+        }
+        [Fact]
+        public void TrackMediaEntitiesShouldBePopulatedWithMediaEntities()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { NonTrackingContext }, true);
+
+            var documents = NonTrackingContext.Documents.ExecuteAsync().GetAwaiter().GetResult().ToList();
+
+            Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.ToList().Count);
+
+            // verify that the stream links are equal 
+            var document = documents.First();
+            var doc = (BaseEntityType)document;
+            Assert.NotNull(doc.EntityDescriptor);
+            Assert.NotNull(NonTrackingContext.GetReadStreamUri(document));
+            Assert.Equal("https://localhost:8000/Documents/1/content", NonTrackingContext.GetReadStreamUri(document).ToString());
+
+            //try and get the content and verify that the content matches the values
+            Stream result = GetTestReadStreamResult(NonTrackingContext, document).GetAwaiter().GetResult();
+
+            Assert.Equal("Hello World!", new StreamReader(result).ReadToEnd());
+
+        }
+
+        private async Task<Stream> GetTestReadStreamResult(DataServiceContext dataServiceContext, Document document)
+        {
+            dataServiceContext.Configurations.RequestPipeline.OnMessageCreating = args =>
+            {
+                // if we read the wrong link then reply with gibberish
+                var resp = (args.RequestUri == document.EntityDescriptor.ReadStreamUri) ? AttachmentResponse : "gibberish";
+
+                return new CustomizedRequestMessage(
+                    args,
+                    resp,
+                    new Dictionary<string, string>()
+                    {
+                            {"Content-Type", "text/plain"},
+                    });
+            };
+
+            var streamResponse = await dataServiceContext.GetReadStreamAsync(document, new DataServiceRequestArgs());
+            return streamResponse.Stream;
+        }
+#if NET452
+        [Fact]
+        public void TestAddingNewItemsBehaviourShouldBeUnAltered()
+        {
+            SetupContextWithRequestPipelineForSaving(
+                new DataServiceContext[] { NonTrackingContext, DefaultTrackingContext }, false);
+
+            var user = new User
+            {
+                Name = "Some name"
+            };
+
+            DefaultTrackingContext.AddObject("Users", user);
+            NonTrackingContext.AddObject("Users", user);
+            Assert.NotNull(NonTrackingContext.GetEntityDescriptor(user));
+            Assert.NotNull(DefaultTrackingContext.GetEntityDescriptor(user));
+
+            SaveContextChanges(new DataServiceContext[] { DefaultTrackingContext, NonTrackingContext });
+            Assert.Equal(1, DefaultTrackingContext.Entities.Count);
+            Assert.Equal(1, NonTrackingContext.Entities.Count);

```

Comment:
Does this mean that the non-tracking context is actually tracking the user entities? (which don't have a stream)

Reply:
Adding should behaviour stays the same as the entities are tracked. but fetched entities are not